src/domain/collection/curatedEventList/CuratedEventList.tsx

```diff
@@ -33,10 +38,25 @@ const CuratedEventList: React.FC<Props> = ({ collection }) => {
     [collection.curatedEvents]
   );
 
-  const { data: eventsData, loading } = useEventsByIdsQuery({
-    variables: { ids: eventIds, include: ['keywords', 'location'] },
+  const queryVariables = {
+    ids: eventIds.slice(0, PAGE_SIZE),
+    include: ['keywords', 'location'],
+  };
+
+  const { data: eventsData, loading, fetchMore } = useEventsByIdsQuery({
+    variables: queryVariables,
   });
 
+  const page = React.useRef(
+    // if eventsByIds is available on first render, they are coming from cache
+    // Initialize page number based on its length
+    eventsData?.eventsByIds.length
+      ? Math.ceil(eventsData?.eventsByIds.length / PAGE_SIZE)
+      : INITIAL_PAGE
+  );
+  const currentEventsCount = page.current * PAGE_SIZE;
```

Comment:
Is this more like `eventCursor` or `eventCursorIndex` or something?

Reply:
is it? I don't know the is the right term here

Reply:
eventCursorIndex sounds good to me :)